### DISCUSSION TITLE
Esphome mancato flash dispositivi via seriale (An error occurred. Improv Wi-Fi Serial not detected?

### DISCUSSION THREAD

**Discussion by dante70** [indomus/forum#2637] [Hardware & Firmware]
salve a tutti,
da un paio di anni uso esphome e faccio flash di  dispositivi collegandomi con convertitore seriale usb e fisicamente saldando cavetti a bordo dei dispositivi(esempio:son off mini,dual r3,mini r4).
Ho sempre usato esphome flasher che sulla finestra mi da l'andamento del flash e poi appaiono i log.
da un po' di tempo a questa parte si puo' usare esphome web.
Mi accorgo la settimana scorsa che non riesco a flashare piu nessun dispositivo ne con esphome flasher ne da web.
Con esphome flasher il flash sembra funzionare ma poi non appaiono log e quindi il dispositivo resta scollegato
Da esphome web si collega al dispositivo ,carica ma poi appare "An error occurred. Improv Wi-Fi Serial not detected".
praticamente mi semmbra di interpretare che non si attiva wi fi del dispositivo e quindi non si puo proseguire immettendo credenziali della propria rete.
girovagando in rete mi e' parso di intendere che potesse esssere un problema del convertitore e relativi driver....ho provato usando arduino uno come seriale e il risultato e' il medesimo.
collegando direttamente convertitore e dispositivo da flashare al Raspberry dove c'è l'istallazione di HA non cambia nulla.
eseguendo flash su dispositivi D1 con usb non ci sono problemi.
Deduco che il problema stia nel collegamento seriale..................che cavolo e' cambiato da qualche tempo a questa parte?

ringrazio chi possa aiutarmi

**Comment by coccovax** [comment]
Ho lo stesso problema che descrivi tu. Sei riuscito a risolvere? Se sì, potresti cortesemente indicare la soluzione?

**Reply by dante70** [reply]
ciao
io dopo tanto tempo senza poter fare flash qualche mese orsono ho riprovato con un tool trovato in rete e son riuscito.
Trattasi di : Adafruit ESPTool.
Al momento non so dirti altro.Son mesi che non ho tempo per domotica ma ti assicuro che ho fatto flash di son off dual r3 e mini r4 che avevo tenuto per un anno e piu' smantellati in una scatola
